### PR TITLE
Add the watchDirectory method to Inotify

### PR DESCRIPTION
**Pull request by harmjan**
I needed to be able to get the events from a directory and realized a bit late that that method wasn't available so I added it myself. I think I kept the code style pretty similar to what was already there. If there was another way to get the events of a directory directly I would also like to hear that.

I would be open to processing some comments if you have them.

### CODE REVIEW

File: source/inotify.cpp [hunk 1]
```diff
@@ -76,6 +76,32 @@ void Inotify::watchFile(const FileSystemEvent& fse)
...
 }
 
+void Inotify::watchDirectory(const FileSystemEvent& fse)
+{
+    if (!checkWatchDirectory(fse))
+        return;
+
+    mError = 0;
+    int wd = 0;
+    wd = inotify_add_watch(mInotifyFd, fse.getPath().c_str(), getEventMask(fse.getEvent()));
```
Comment (sizeofvoid) [inline source/inotify.cpp]:
```suggestion
    const int wd = wd = inotify_add_watch(mInotifyFd, fse.getPath().c_str(), getEventMask(fse.getEvent()));
```

**Review by sizeofvoid — CHANGES_REQUESTED** [review]
Unfortunately I am currently not able to test this.  From 19.08 I will take care of it. Thanks for the feature and the PR

**Comment by erikzenker** [comment]
@harmjan could also make sense to add the function to the original upstream repo

File: source/inotify.cpp (hunk 1 above)
Comment (harmjan) [reply source/inotify.cpp]: I replaced it in 6886f46, I copied that bit from `watchFile` though so I also changed it there.

**Pull request merged**